Client/Application/AppDelegate.swift

```diff
@@ -81,6 +82,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject?) -> Bool {
+        if url.scheme == "firefox" {
+            if url.lastPathComponent == "open-url" {
+                if let query = url.query {
+                    if let index = query.unescape().rangeOfString("=") {
```

Comment:
Instead of doing this with `rangeOfString`, use the `NSURLComponents` class. That will take a string and will parse the URL and and query for you.